### PR TITLE
Add staff assignment management to enrollment overview

### PR DESCRIPTION
**Pull request by martha**
## Description

[//]: # 'remove if not applicable'
Depends on hmis-warehouse PR: https://github.com/greenriver/hmis-warehouse/pull/4564
GH issue: https://github.com/open-path/Green-River/issues/6335
Figma: https://www.figma.com/design/F9EHR2IZYuMAlOJ9aOG2IL/OP-%2F-HMIS-%2F-2024-%2F-Q1-%2B-Q2-(Jan-1---June-30)?node-id=417-1961&t=go59iBv2Tl8nSyW9-0

## Type of change
[//]: # 'remove options that are not relevant'
- [ ] Bug fix
- [x] New feature (adds functionality)
- [ ] Code clean-up / housekeeping
- [ ] Dependency update

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (eslint)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: src/modules/enrollment/components/staffAssignment/StaffAssignmentCard.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+}) => {
+  const { enrollment } = useEnrollmentDashboardContext();
+
+  const {
+    data: { household } = {},
+    loading,
+    error,
+  } = useGetHouseholdStaffAssignmentsQuery({
+    variables: { id: householdId },
+  }); // fixme - this query gets repeated twice with the same variables; why?
```
Comment (martha) [inline src/modules/enrollment/components/staffAssignment/StaffAssignmentCard.tsx]: @ttoomey when you review this, could you take a look at this and let me know if you have any ideas to improve it? I'm flummoxed - even memoizing the `StaffAssignmentCard` component on the `householdId` does not seem to fix it.

Network tab:
3rd query includes `GetHouseholdStaffAssignments`
Screenshot 2024-07-26 at 8 43 36 AM

4th query asks for the same data, even though the component is already loaded and visible:
Screenshot 2024-07-26 at 8 44 20 AM

Comment (ttoomey) [reply src/modules/enrollment/components/staffAssignment/StaffAssignmentCard.tsx]: @martha Thanks for finding this repeated query!

It was indeed a bit tricky to track this down. So I believe what's happening is that another query, `GetHouseholdClientAlerts`, is running on the same page; this query seems to be missing the id field on the household. This is important because Apollo's query cache uses the id (or configured keyFields) to determine how to cache the results.

When this second query resolves, it seems like the behavior is that since Apollo can't determine how to cache the query results, it reruns the initial household query. I'm not sure if this is a bug or intentional behavior in Apollo itself, maybe it's trying to be safe and just invalidating the household cache.

Generally all queries should resolve the id or key fields. Usually this is part of the fragment already but in this case they need to be included explicitly.

This seems to fix the issue for me locally:
```diff
diff --git a/src/api/operations/alert.queries.graphql b/src/api/operations/alert.queries.graphql
index 19462745..4e2e805f 100644
--- a/src/api/operations/alert.queries.graphql
+++ b/src/api/operations/alert.queries.graphql
@@ -1,6 +1,8 @@
 query GetHouseholdClientAlerts($id: ID!) {
   household(id: $id) {
+    id
     householdClients {
+      id
       client {
         ...ClientWithAlertFields
       }
diff --git a/src/types/gqlTypes.ts b/src/types/gqlTypes.ts
index b10c2521..32e3db47 100644
--- a/src/types/gqlTypes.ts
+++ b/src/types/gqlTypes.ts
@@ -8189,8 +8189,10 @@ export type GetHouseholdClientAlertsQuery = {
   __typename?: 'Query';
   household?: {
     __typename?: 'Household';
+    id: string;
     householdClients: Array<{
       __typename?: 'HouseholdClient';
+      id: string;
       client: {
         __typename?: 'Client';
         id: string;
@@ -35089,7 +35091,9 @@ export type DeleteClientAlertMutationOptions = Apollo.BaseMutationOptions<
 export const GetHouseholdClientAlertsDocument = gql`
   query GetHouseholdClientAlerts($id: ID!) {
     household(id: $id) {
+      id
       householdClients {
+        id
         client {
           ...ClientWithAlertFields
         }
```

File: src/modules/enrollment/components/staffAssignment/EditStaffAssignmentDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    );
+  }, [household.householdClients]);
+
+  const { enrollment } = useEnrollmentDashboardContext();
+
+  if (!enrollment) return;
+  if (!headOfHousehold) return;
+
+  return (
+    <CommonDialog fullWidth open={open} onClose={onClose}>
```
Comment (ttoomey) [inline src/modules/enrollment/components/staffAssignment/EditStaffAssignmentDialog.tsx]: We have so much crammed in this one modal. Maybe we should make it a little wider?
```suggestion
    <CommonDialog fullWidth open={open} onClose={onClose} maxWidth='md'>
```

File: src/modules/enrollment/components/staffAssignment/EditStaffAssignmentDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+      <DialogTitle>Manage Staff Assignments</DialogTitle>
+      <DialogContent>
+        <Stack gap={2} sx={{ mt: 2 }}>
+          <Stack direction={isTiny ? 'column' : 'row'} gap={isTiny ? 2 : 4}>
```
Comment (ttoomey) [inline src/modules/enrollment/components/staffAssignment/EditStaffAssignmentDialog.tsx:53]: maybe a little more space here?
```suggestion
          <Stack direction={isTiny ? 'column' : 'row'} gap={isTiny ? 2 : 4} sx={{my:1}}>
```

File: src/modules/enrollment/components/staffAssignment/UnassignStaffButton.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    refetchQueries: [
+      GetHouseholdStaffAssignmentsDocument,
+      GetHouseholdStaffAssignmentHistoryDocument,
+    ],
+    awaitRefetchQueries: true,
+  });
+
+  if (error) throw error;
+
+  return (
```
Comment (ttoomey) [inline src/modules/enrollment/components/staffAssignment/UnassignStaffButton.tsx]: I wonder if we should confirm this action?

File: src/modules/enrollment/components/staffAssignment/EditStaffAssignmentDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+              rows={household.staffAssignments?.nodes || []}
+              columns={[
+                STAFF_ASSIGNMENT_COLUMNS.staffName,
+                STAFF_ASSIGNMENT_COLUMNS.role,
+                STAFF_ASSIGNMENT_COLUMNS.assignmentDate,
+                STAFF_ASSIGNMENT_COLUMNS.action,
+              ]}
+              noData={'No Staff Assigned'}
+              tableProps={{ 'aria-label': 'Current staff assignments' }}
+            />
```
Comment (ttoomey) [inline src/modules/enrollment/components/staffAssignment/EditStaffAssignmentDialog.tsx]: This is fine as-is but maybe it would be a nicer to extract into another component, StaffAssignmentTable, along with the STAFF_ASSIGNMENT_COLUMNS constant.

File: src/modules/enrollment/components/staffAssignment/StaffAssignmentCard.tsx [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+        {!loading && household?.staffAssignments && (
+          <>
+            <Divider />
+            {household.staffAssignments.nodesCount === 0 && (
+              <Typography sx={{ p: 2 }} color='text.secondary'>
+                No Staff Assigned
+              </Typography>
+            )}
+            <CommonDetailGridContainer>
+              {Object.entries(assignmentsByType).map(([type, assignments]) => (
```
Comment (ttoomey) [inline src/modules/enrollment/components/staffAssignment/StaffAssignmentCard.tsx]: This seems fine for mvp but I have a couple of concerns
* it's not clear to me that the assignments will be ordered in a consistent way. Maybe they are?
* I'm not sure about the value of grouping assignments by role rather than just showing each staff assignment and role on their own row. Seems like there's some value in having the presentation be consistent
* One possibility would be to try and re-use our the "Current Staff Assignments" table from the modal, perhaps with a 'read-only' variant to hide the 'unassign' column

File: src/modules/enrollment/components/staffAssignment/EditStaffAssignmentDialog.tsx [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+            </CommonLabeledTextBlock>
+          </Stack>
+          <NewStaffAssignmentForm
+            householdId={household.id}
+            projectId={enrollment.project.id}
+          />
+          <Typography
+            id='current-staff-assignments'
+            sx={{ mt: 1 }}
+            variant='h5'
```
Comment (ttoomey) [inline src/modules/enrollment/components/staffAssignment/EditStaffAssignmentDialog.tsx:71]: To me, this title feels like it's too large but perhaps I'm just reacting to having so much in one modal. Also I don't know that I can suggest another typographic variant that would be more suitable so maybe we can leave it for now.

**Review by ttoomey — APPROVED** [review]
Looks great, thanks!

I left a few suggestions that are optional

File: src/modules/enrollment/components/staffAssignment/StaffAssignmentCard.tsx (hunk 1 above)
Comment (martha) [reply src/modules/enrollment/components/staffAssignment/StaffAssignmentCard.tsx]: Stellar, thank you so much @ttoomey for the sleuthing and for the thorough explanation too!

File: src/modules/client/components/clientAlerts/ClientAlertCard.tsx [hunk 8]
```diff
@@ -30,32 +30,34 @@ const ClientAlertCard: React.FC<ClientAlertCardProps> = ({
...
-            {children}
-          </Stack>
-        )}
-        {clientAlerts.length > 0 && (
-          <ClientAlertStack clientAlerts={clientAlerts}>
-            {children}
-          </ClientAlertStack>
-        )}
-      </Box>
+      {loading && <Loading />}
```
Comment (martha) [inline src/modules/client/components/clientAlerts/ClientAlertCard.tsx:33]: This is an unrelated style change for consistency with the other components loading on this page. The contents of the component are not changing, it's just moving the Loading indicator outside the Box, and wrapping the whole Box in a conditional so it only shows when loading is false.

before:
Screenshot 2024-07-29 at 8 36 32 AM (2)

after:
Screenshot 2024-07-29 at 8 40 36 AM (2)

File: src/modules/enrollment/components/staffAssignment/EditStaffAssignmentDialog.tsx (hunk 3 above)
Comment (martha) [reply src/modules/enrollment/components/staffAssignment/EditStaffAssignmentDialog.tsx:53]: I like the idea of adding more space, maybe as added gap in the Stack rather than margin on just one of the Stack elements, so it's applied consistently?

File: src/modules/enrollment/components/staffAssignment/UnassignStaffButton.tsx (hunk 4 above)
Comment (martha) [reply src/modules/enrollment/components/staffAssignment/UnassignStaffButton.tsx]: Let's discuss at design session tomorrow? Since this is all already happening within a modal, it might not make sense to use our usual confirmation modal pattern.

Figma thread https://www.figma.com/design/F9EHR2IZYuMAlOJ9aOG2IL?node-id=4706-52377#891432907

File: src/modules/enrollment/components/staffAssignment/EditStaffAssignmentDialog.tsx (hunk 5 above)
Comment (martha) [reply src/modules/enrollment/components/staffAssignment/EditStaffAssignmentDialog.tsx]: Good plan 👍

File: src/modules/enrollment/components/staffAssignment/StaffAssignmentCard.tsx (hunk 6 above)
Comment (martha) [reply src/modules/enrollment/components/staffAssignment/StaffAssignmentCard.tsx]: Yeah, I was thinking about this as I was writing it too. Let's raise it in the design session tomorrow.

Figma thread: https://www.figma.com/design/F9EHR2IZYuMAlOJ9aOG2IL?node-id=4701-47650#876723704

File: src/modules/enrollment/components/staffAssignment/EditStaffAssignmentDialog.tsx (hunk 7 above)
Comment (martha) [reply src/modules/enrollment/components/staffAssignment/EditStaffAssignmentDialog.tsx:71]: Ok sounds good. Thanks very much @ttoomey for your thorough review and all your comments, I really appreciate it!

File: src/modules/enrollment/components/staffAssignment/StaffAssignmentTable.tsx [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+  },
+  assignmentDateRange: {
+    header: 'Assignment Dates',
+    render: (assignment) =>
+      parseAndFormatDateRange(assignment.assignedAt, assignment.unassignedAt),
+  },
+  action: {
+    header: 'Action',
+    render: (assignment) => (
+      <DeleteMutationButton<
```
Comment (martha) [inline src/modules/enrollment/components/staffAssignment/StaffAssignmentTable.tsx:45]: Pending team discussion tomorrow at design session https://www.figma.com/design/F9EHR2IZYuMAlOJ9aOG2IL?node-id=4706-52377#891432907
